workflow/controller/indexes/workflow_index.go

```diff
@@ -70,6 +71,16 @@ func WorkflowSemaphoreKeysIndexFunc() cache.IndexFunc {
 		if err != nil {
 			return nil, nil
 		}
-		return wf.GetSemaphoreKeys(), nil
+		keys := wf.GetSemaphoreKeys()
+		// store all keys while indexing as a side effect
+		for _, key := range keys {
+			semaphoreKeyMap[key] = struct{}{}
```

Comment:
This is pretty hacky, I don't like this, but every way to track this seemed hacky 😕
This option had the least changes though, which is why I ended up going with it